### ISSUE TITLE
custom YOLOV8n object detection model not working

### ISSUE THREAD

**Issue by MichaelRinger** [zezo357/pytorch_lite#12]
Hello,

I trained a custom YOLOV8n-model in google colab for object detection, exported it to torchscript, but it doesnt return any predictions when using it in flutter.

Might be the pytorch version to high, that has been used to train that model because I got this in the debug console:
![image](https://user-images.githubusercontent.com/33952961/222974357-40d5bf5f-26a6-4135-ac3c-a01f0cca8006.png)

Training commands:
```
rf = Roboflow(api_key="-----------")
project = rf.workspace("karel-cornelis-q2qqg").project("aicook-lcv4d")
dataset = project.version(4).download("yolov8")
!yolo train task=detect model={MODEL_PATH} data={dataset.location}/data.yaml project={SAVE_DIR} epochs=100 imgsz=640 batch=64
!yolo mode=export model={SAVE_DIR}/train/weights/best.pt format=torchscript optimize
```

The unoptimized model worked in colab.

**Comment by zezo357** [comment]
Can you try different values for detection iou and score starting value.

And if it didn't work can you send the model and image of it

**Comment by MichaelRinger** [comment]
I now tested the model with the camera demo and found out that it works, but just very poorly.
Thanks for the fast reply.

**Comment by MichaelRinger** [comment]
Is it normal that the torchscript version of a yolo model less accurate is? I exported the normal yoloV8n.pt model with 23 MB to a torchscript model that has only 11 MB and get way less accuracy even when using the extra optimize flag in the yolo cli.

**Comment by zezo357** [comment]
> Is it normal that the torchscript version of a yolo model is less accurate? I exported the normal yoloV8n.pt model with 23 MB to a torchscript model that has only 11 MB and has  a way worse accuracy even when using the extra optimize flag in the yolo cli.

Yes the torch script should be less accurate but   in an acceptable manner.

Don't know why it may be way worse.


Are you checking with the same inputs ?(image, iou, threshold) on both? And if yes can you share your results?

**Comment by MichaelRinger** [comment]
The torchscript model in colab worked a lot better.

### torchscript model in google colab:
![image1](https://user-images.githubusercontent.com/33952961/223182631-0d25e960-8750-45c5-bd53-e4ef342ae69f.jpg)

**code:**
```
from ultralytics import YOLO
from PIL import Image
from google.colab.patches import cv2_imshow

img = '/content/drive/MyDrive/SnapMenu/2020-3260-6.jpg'
model = YOLO(SAVE_DIR+'/train/weights/best.torchscript', task='detect')
res = model(img)
res_plotted = res[0].plot()
cv2_imshow(res_plotted)
```



### torchscript model in flutter (pytorch_lite):
![image](https://user-images.githubusercontent.com/33952961/223182658-89e6f330-8bfc-4531-b99e-ee28b57bf0c2.jpg)

**code:**
tested in the pytorch_lite example app, by just changing the yolov8 model and classes file.

**Comment by zezo357** [comment]
Can you upload the model and image? I want to test to make sure where is the problem


Torchscript, or my implementation

**Comment by MichaelRinger** [comment]
You can download the model and the sample image from here:
https://filetransfer.io/data-package/2WjiAi5q#link

I tested the yolov8s model that comes with your example flutter application and it works also quite poorly, so it might be actually the implementation

**Comment by zezo357** [comment]
> You can download the model and the sample image from here:
> https://filetransfer.io/data-package/2WjiAi5q#link
> 
> I tested the yolov8s model that comes with your example flutter application and it works also quite poorly, so it might be actually the implementation. But the yolov5s-model seems to work fine.

Yolov5 I followed the official example 

But for yolov8 I followed an adaptation in a branch so it may have a problem I didn't notice 


This is why I will test again to be sure

**Comment by MichaelRinger** [comment]
The yolov8s only seems to detect persons and no other classes inside the application. Hope that information might help.

**Comment by zezo357** [comment]
@MichaelRinger 

can you test 2.0.3?

it should be fixed now

**Comment by zezo357** [comment]
will wait until you confirm that its working as expected

**Comment by MichaelRinger** [comment]
I changed the code in the Preprocessor.java to test it. I changed the following:

```
// mOutputColumn = 8;
mOutputColumn = (mNumberOfClasses + 4);
```

The results are better but still not as good as the google colab inference result:
![test_image](https://user-images.githubusercontent.com/33952961/223212781-d4bcd59a-09d4-48f3-a3fc-9ffce54f24a3.jpg)

**Comment by zezo357** [comment]
> I changed the code in the Preprocessor.java to test it. I changed the following:
> 
> ```
> // mOutputColumn = 8;
> mOutputColumn = (mNumberOfClasses + 4);
> ```
> 
> The results are better but still not as good as the google colab inference result: ![test_image](https://user-images.githubusercontent.com/33952961/223212781-d4bcd59a-09d4-48f3-a3fc-9ffce54f24a3.jpg)

yes i did this change in my release

now the results are more logical, yes lower than the full model but gets most of the values 


now maybe trying a different IOU gets you better results

**Comment by MichaelRinger** [comment]
Changing the IOU and score didnt change much. I tried to replicate the test on the native android pytorch object detection implementation, but I stumbled upon some errors that might be due to the fact, that the implementation is only with yoloV5 compatible. Do you know what I need to change in order get yoloV8 working the android object detection example.

android object detection:
https://github.com/pytorch/android-demo-app/tree/master/ObjectDetection

btw. I discoverd a typo. You wrote IOUThershold instead of IOUThreshold in your plugin.

**Comment by zezo357** [comment]
> Changing the IOU and score didnt change much. I tried to replicate the test on the native android pytorch object detection implementation, but I stumbled upon some errors that might be due to the fact, that the implementation is only with yoloV5 compatible. Do you know what I need to change in order get yoloV8 working the android object detection example.
> 
> android object detection:
> https://github.com/pytorch/android-demo-app/tree/master/ObjectDetection
> 
> btw. I discoverd a typo. You wrote IOUThershold instead of IOUThreshold in your plugin.

I fixed the typo in 2.0.5 


For the android native one moment I will give you the link of the branch

**Comment by zezo357** [comment]
This is the branch I followed: https://github.com/atanasko/android-demo-app
